### ISSUE TITLE
Source-google-ads: Optionally skip records on same day as the input cursor

### ISSUE THREAD

**Issue by girarda** [airbytehq/airbyte#29814]
## What
Each sync, the connector syncs data from [`initial_cursor_date`, `today`], inclusively, and persists `today` as the new `initial_cursor_date`.

This will lead to the records of `today` being duplicated on the next sync. 

This is a problem for users that sync data in `incremental | append`, because they see duplicate records in their destination.

One solution to the problem is to add a config option to only sync data exclusively more recent than the `initial_cursor_date`.

## Acceptance criteria
- [ ] It is possible for users to configure the connector to sync data from `]start_time, end_time]`
- [ ] The default behavior is the existing one: sync data from `[start_time, end_time]`
- [ ] When running an initial sync or a full refresh, the config's `start_time` is inclusive
- [ ] The sync's start_time is defined as the `initial_cursor_date` - `lookback window`

**Comment by sherifnada** [comment]
cc @tolik0 

could this cause data loss? if GAds has a daily granularity, and the following syncs happen: 

sync #1 is run at 1pm on october 10. It syncs from `oct1` to `oct10`
sync #2 is run at 1pm on october 11. It syncs from `oct 11` only. 

Do we lose the data from 1pm-midnight for `oct10`? 

as long as we protect against this situation this seems like a useful option to add.

**Comment by tolik0** [comment]
@sherifnada Indeed, after the sync is completed, we will miss all data for October 10. We can approach this issue in two ways:

**Sync with Previous Day's End Date:** This method ensures that all records up to the previous day are synced, avoiding any duplication.

sync #1 (Oct 10): Data from Oct 1 to Oct 9 (All date ranges are inclusive).
sync #2 (Oct 10): No data to sync. 
sync #3 (Oct 11): Data for Oct 10.

By adopting this strategy, we prevent creating duplicates, but there will be a one-day delay in data availability.

**Sync with Today's End Date:** This method syncs up to the current date, but it may result in duplicate records during multiple syncs within a day.

sync #1 (Oct 10): Data from Oct 1 to Oct 10.
sync #2 (Oct 10): Data for Oct 10.
sync #3 (Oct 11): Data from Oct 10 to Oct 11.

While this approach provides the most up-to-date data, it comes at the expense of duplicating records for the stream state granularity, which is one day in our scenario.

This challenge is prevalent in components whose granularity isn't small enough, leading to cursor value collisions and subsequent data duplication.
